interfaces/cython/cantera/composite.py

```diff
@@ -682,6 +684,8 @@ def append(self, state=None, normalize=True, **kwargs):
         if state is not None:
             self._phase.state = state
 
+        # Non normalize form must be set using the mass fractions as the
```

Comment:
@bryanwweber, thank you for pointing the `Near` issue. It uncovered another, more fundamental issue. Since "Y" is the native state parameter, 'non-normalized` forms should be appended using that.

Reply:
I forgot to check that.  It's not normalized.

Reply:
Thanks for the clarification! I'm not sure what's the default, but I believe append with `state` should *not* normalize as this would preserve the true value by default (i.e. `normalize=False` should be set).

Reply:
Looks like the logic actually has to be a little more complex, as a user would expect `normalize=True` if a composition is appended, whereas `normalize=False` would be the expectation if `state` or a full numpy array is used for either `X` or `Y`.

Reply:
I don't understand the need for this change... `set_mole_fractions_unnormalized` sets the mass fractions to the values corresponding to the appropriate unnormalized mole fractions. Can you clarify?

Reply:
I believe values were sufficiently close in the `assertNear` unit test that it masked the fact that the test actually failed. @DavidAkinpelu ... could you confirm?

Reply:
> I don't understand the need for this change... `set_mole_fractions_unnormalized` sets the mass fractions to the values corresponding to the appropriate unnormalized mole fractions. Can you clarify?

It is true `set_mole_fractions_unnormalized` sets mass fractions to the corresponding values without normalizing. The only concern is that `mole_fractions` and `mass_fractions` calculates `mean_molecular_weight` differently which causes discrepanies if we used `set_mole_fractions_unnormalized` and the resulting values are normalized to 1.0.